File: Codigo/Backend/PharmaGo.Domain/Entities/Product.cs
using PharmaGo.Domain.Entities;
using PharmaGo.Exceptions;
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace PharmaGo.Domain.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public Pharmacy? Pharmacy { get; set; }
        public bool Deleted { get; set; }

        public void ValidOrFail()
        {
            if (string.IsNullOrEmpty(Code) || string.IsNullOrEmpty(Name) || string.IsNullOrEmpty(Description)
                    || Price <= 0 || Pharmacy == null)
                throw new InvalidResourceException("Mandatory information is missing.");
            if (!Regex.IsMatch(Code, @"^\d{5}$"))
                throw new InvalidResourceException("The product code is invalid.");
            if (Name.Length > 30)
                throw new InvalidResourceException("The product name is too long.");
            if (Description.Length > 70)
                throw new InvalidResourceException("The product description is too long.");
        }
    }
}

File: Entrega 5/Codigo/Frontend/src/app/interfaces/product.ts
export interface Product {
    id: number;
    code: string;
    name: string;
    description: string;
    price: number;
    quantity: number;
    pharmacy: {
        id: number;
        name: string;
    };
}

export class ProductRequest {
    code: string;
    name: string;
    description: string;
    price: number;
    quantity: number;
    pharmacyName: string = "";

    constructor(code: string, name: string, description: string, price: number, quantity: number, pharmacyName: string) {
        this.code = code;
        this.name = name;
        this.description = description;
        this.price = price;
        this.quantity = quantity;
        this.pharmacyName = pharmacyName;
    }
}
